### PR TITLE
Fix bug in loading custom notification schedule config.

### PR DESCRIPTION
**Pull request by demiankatz**
The processing of the scheduled_search_frequencies setting does not appear to be working correctly in all situations, which sometimes results in a failure of the command line tool for sending notifications (complaining about an object being passed where an array is expected). This PR makes the processing more explicit and ensures that the setting is always an array.

### CODE REVIEW

File: module/VuFind/src/VuFind/Search/History.php [hunk 1]
```diff
@@ -136,7 +138,12 @@ public function getScheduleOptions()
         if (!($this->config->Account->schedule_searches ?? false)) {
             return [];
         }
-        return $this->config->Account->scheduled_search_frequencies
-            ?? [0 => 'schedule_none', 1 => 'schedule_daily', 7 => 'schedule_weekly'];
+        if ($this->config->Account->scheduled_search_frequencies instanceof Config) {
```
Comment (EreMaijala) [inline module/VuFind/src/VuFind/Search/History.php]: Is it ok to check instanceof with a potentially missing element? I suppose this works most of the time as the Account section is usually defined, but if it wasn't, wouldn't this balk? So it would be safer to move this check inside the isset block below?

Comment (demiankatz) [reply module/VuFind/src/VuFind/Search/History.php]: If the account section isn't set, then the check for scheduled_searches will evaluate to false, and the code will never get this far. I think that makes this safe, though perhaps in need of a clarifying comment. However, your observation also reminds me that our test coverage for this method is not complete, since otherwise this bug would never have gotten through in the first place... thus, I'll expand the tests today to make sure all the edge cases are checked, and make adjustments as needed.

Comment (EreMaijala) [reply module/VuFind/src/VuFind/Search/History.php]: You're right. There's still a chance that a later change could introduce a bug since it's sort of a side-effect, but I suppose it's fine.

**Comment by demiankatz** [comment]
@EreMaijala, I've added the missing tests and reformatted the code for improved clarity. Mind taking one last look? Thanks!

File: module/VuFind/src/VuFind/Search/History.php [hunk 2]
```diff
@@ -133,10 +135,19 @@ public function getSearchHistory($userId = null)
      */
     public function getScheduleOptions()
     {
+        // If scheduled searches are disabled, return no options:
         if (!($this->config->Account->schedule_searches ?? false)) {
             return [];
         }
-        return $this->config->Account->scheduled_search_frequencies
-            ?? [0 => 'schedule_none', 1 => 'schedule_daily', 7 => 'schedule_weekly'];
+        // If custom frequences are not provided, return defaults:
```
Comment (EreMaijala) [inline module/VuFind/src/VuFind/Search/History.php]: Typo: frequences

**Review by EreMaijala — APPROVED** [review]
Looks good, apart from a small comment typo.

**Comment by demiankatz** [comment]
Thanks for the catch, @EreMaijala! Typo fixed, so merging now!
